### ISSUE TITLE
Wrong computation time for multicore execution after a sleep

### ISSUE THREAD

**Issue by maelmadon** [oar-team/batsim#66]
Hello, 
Today I bumped into an issue with my experiments, I don't know yet if it's a bug from my side, Batsim or Simgrid.

## Bug description 
I have a two identical jobs with 4 parallel tasks each, submitted at t=0 and t=5000.

- Case 1: I schedule job0 and job1 on the multicore machine0, staying idle between executions 
![image](https://user-images.githubusercontent.com/36968988/133270352-09668cbf-af8f-425a-b9de-7ad429b3c03f.png)

- Case 2: after job0 is finished I switch off machine0, and switch it back on at t=5000 to run job1
![image](https://user-images.githubusercontent.com/36968988/133271157-e8706ddb-f46d-46f6-bda3-841d28e6d5b7.png)

**Expected behavior:** job0 and job1 should have the same execution time. In fact, here, job1 takes exactly 4 times longer than job0, which corresponds to its number of parallel executors. As if machine0 was not running multicore anymore after rebooting...

## Versions  
- Batsim commit: commit 0e24a9009d317adc94324d058e1caed58746fe56 (built by Nix from master branch)
- SimGrid version: 3.28.0

## Logs  
[batsim.log](https://github.com/oar-team/batsim/files/7162532/batsim.log) with verbosity debug for this experiment.

**Comment by mpoquet** [comment]
Mmh I confirm that I can see the same thing that you described by reading Batsim logs.

- The two jobs seems to use the same profile.
- The two jobs are allocated on the same host, which seems to be on the same state (nothing running on it) and on the same pstate (0)
- First job completes in 3520 s, while the second completes in 14080=4*3520 s.

Potential issues:
- Are your machines initialized with pstate=0 in your platform file?
- Is your profile doing communications? If so, the slowdown may come from a different load on network links when the jobs are executed

Can you try to write a small SimGrid code to detect if the error comes from SimGrid or Batsim?
You can for example use mwe.cpp from [SimGrid's issue 37](https://framagit.org/simgrid/simgrid/-/issues/37) as a base.

**Comment by maelmadon** [comment]
Hi @mpoquet and thanks for your answer. My platform file and workload file are pretty straightforward, I pasted them at the end of this post.

- Yes, my machines are initialized with pstate=0 in my platform file.
- My tasks do no communication.

I will try to identify if the problem comes from Simgrid or Batsim and follow up. 

Workload file:
```
{
    "description": "Test binpacking.",
    "nb_res": 12, 
    "jobs": [
        {"id": "0", "profile": "blast_vm_large", "res": 4, "subtime": 0},
        {"id": "1", "profile": "blast_vm_large", "res": 4, "subtime": 5000}
    ],
    "profiles": {
        "blast_vm_large": {"com": 0.0, "cpu": 1.657216e14, "type": "parallel_homogeneous_total"}
    }
}
```

Platform file:
```xml
<?xml version='1.0'?>
<!DOCTYPE platform SYSTEM "http://simgrid.gforge.inria.fr/simgrid/simgrid.dtd">
<platform version="4.1">
<config id="General">
        <prop id="contexts/stack-size" value="16"></prop>
        <prop id="contexts/guard-size" value="0"></prop>

</config>

<zone id="toy_g5k"  routing="Full">
    <host id="master_host" speed="100Mf">
        <prop id="wattage_per_state" value="100:100:200" />
        <prop id="wattage_off" value="10" />
    </host>

        <host id="taurus_0" core="12" pstate="0" speed="11.77Gf, 1e-9Mf, 0.166666666666667f, 0.006666666666667f">
            <prop id="wattage_per_state" value="100:100:217, 9.75:9.75:9.75, 100:100:100, 125:125:125" />
            <prop id="wattage_off" value="10" />
            <prop id="sleep_pstates" value="1:2:3" />
        </host>
        <host id="taurus_1" core="12" pstate="0" speed="11.77Gf, 1e-9Mf, 0.166666666666667f, 0.006666666666667f">
            <prop id="wattage_per_state" value="100:100:217, 9.75:9.75:9.75, 100:100:100, 125:125:125" />
            <prop id="wattage_off" value="10" />
            <prop id="sleep_pstates" value="1:2:3" />
        </host>
</zone>
</platform>
```

**Comment by maelmadon** [comment]
Well intuited @mpoquet, it comes from simgrid. 

I submitted the issue on SimGrid's repo : [issue #95](https://framagit.org/simgrid/simgrid/-/issues/95).

**Comment by Mommessc** [comment]
It seems the SimGrid issue was fixed, can we close this one too?

**Comment by mpoquet** [comment]
I also think that SimGrid#95 fixed this issue, but the following should be done before marking this issue as closed.
- [ ] make sure that SimGrid#95 fixed this specific problem
- [ ] make sure that the latest Batsim release on NUR-Kapack uses a fixed SimGrid version.

NB: A new SimGrid version should be released very soon.

**Comment by maelmadon** [comment]
I can confirm that this bug is fixed with batsim 4.1.0 and simgrid 3.31.0, which is the latest Batsim release on NUR-Kapack.
